### PR TITLE
[Perf] Avoid React.Children.count traversal for empty state early return

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Avoid calls to `React.Children.count` which recursively traverses child components. The cost grows with component hierarchy. These cases were for early return `null` so they should be quick. In cases where `children` are not passed via props the value is `undefined`.

`Text` for example:

image

#### Focus areas to test

(optional)

Related: #15462 #15473

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
These need to explicitly check `=== undefined`. I'll push an iteration. The reason is `<div>{0}</div>` is treated as falsey otherwise (thanks unit tests).

File: packages/react-cards/src/components/Card/CardSection/CardSection.view.tsx
Comment (xugao) [inline packages/react-cards/src/components/Card/CardSection/CardSection.view.tsx:8]:
```suggestion
  if (children == null) {
```

Comment (xugao) [reply packages/react-cards/src/components/Card/CardSection/CardSection.view.tsx:8]: we should do the check like this so it covers both `undefined` and `null`

Comment (KevinTCoughlin) [reply packages/react-cards/src/components/Card/CardSection/CardSection.view.tsx:8]: Sure, tho I wouldn't except `null` ever to be passed. React passes `undefined`. `null` would only be passed by library consumers.

**Review by xugao — APPROVED** [review]

**Comment by khmakoto** [comment]
Let's wait until after the snap period to merge or spin a v7 version of this PR if needed sooner.

**Review by Ridvanovskyy — APPROVED** [review]

File: change/@fluentui-react-internal-2020-10-13-16-47-27-keco-react-children-for-empty.json
Comment (ecraig12345) [inline change/@fluentui-react-internal-2020-10-13-16-47-27-keco-react-children-for-empty.json:2]:
```suggestion
  "type": "prerelease",
```

File: change/@uifabric-react-cards-2020-10-13-16-47-27-keco-react-children-for-empty.json
Comment (ecraig12345) [inline change/@uifabric-react-cards-2020-10-13-16-47-27-keco-react-children-for-empty.json:4]:
```suggestion
  "type": "prerelease",
  "comment": "Avoid React.Children traversal for early return cases",
  "packageName": "@fluentui/react-cards",
```

**Review by ecraig12345 — COMMENTED** [review]
Guessing this can go in now? Needs some minor change file updates and a merge with master.

**Comment by ecraig12345** [comment]
Actually both the change files and the conflicts were easy to fix in the web UI, so I took care of that.

**Comment by KevinTCoughlin** [comment]
> Actually both the change files and the conflicts were easy to fix in the web UI, so I took care of that.

Thank you @ecraig12345 🙇‍♂️. @khmakoto can we merge this?

**Review by khmakoto — APPROVED** [review]

**Review by ecraig12345 — APPROVED** [review]

**Review by Ridvanovskyy — COMMENTED** [review]
👍

**Pull request merged**
